src/backend/storeManagers/hyperplay/games.ts

```diff
@@ -1653,6 +1654,82 @@ export async function downloadGameIpdtManifest(
   }
 }
 
+async function checkIfPatchingIsFaster(
+  oldManifestPath: string,
+  newManifestPath: string,
+  gameInfo: GameInfo
+) {
+  // read manifests
+  const oldManifestJson = JSON.parse(readFileSync(oldManifestPath).toString())
+  const newManifestJson = JSON.parse(readFileSync(newManifestPath).toString())
+
+  // compare manifests
+
+  const { compareManifests } = await import('@hyperplay/patcher')
+  const { estimatedPatchSizeInKB } = compareManifests(
+    oldManifestJson,
+    newManifestJson
+  )
+
+  // calc break point % where patching is faster
+  if (
+    gameInfo?.install?.platform &&
+    gameInfo.channels &&
+    gameInfo?.install?.channelName &&
+    Object.hasOwn(gameInfo.channels, gameInfo.install.channelName)
+  ) {
+    const channelName = gameInfo.install.channelName
+    const [releaseMeta] = getReleaseMeta(gameInfo, channelName)
+    const platform = handleArchAndPlatform(
+      gameInfo.install.platform,
+      releaseMeta
+    )
+    const downloadSize = parseInt(
+      releaseMeta.platforms[platform]?.downloadSize ?? '0'
+    )
+    const installSize = parseInt(
+      releaseMeta.platforms[platform]?.installSize ?? '0'
+    )
+    // @TODO: get these speed values from local checks of download/write speed
+    const downloadSpeedInKBPerSecond = 25 * 1024
+    const extractionSpeedInKBPerSecond = 50 * 1024
```

Comment:
open to suggestions here. I'd suggest we leave for another future iteration though